File: src/main/java/com/ada/conversation/infrastructure/in/rest/filter/ValidateChatRequestFilter.java
package com.ada.conversation.infrastructure.in.rest.filter;

import com.ada.conversation.application.dto.ChatRequest; import com.ada.conversation.application.port.in.RequestFilter; import org.springframework.core.annotation.Order; import org.springframework.stereotype.Component;
@Component @Order(20) public class ValidateChatRequestFilter implements RequestFilter { public boolean supports(ChatRequest r){return r.message().isBlank();} public ChatRequest apply(ChatRequest r){throw new IllegalArgumentException("Chat message must not be blank");} }

File: src/main/kotlin/com/ada/conversation/infrastructure/in/rest/filter/ValidateChatRequestFilter.kt
package com.ada.conversation.infrastructure.`in`.rest.filter

import com.ada.dto.ChatRequest
import com.ada.conversation.application.port.`in`.RequestFilter
import org.springframework.core.annotation.Order
import org.springframework.stereotype.Component

@Component
@Order(20)
class ValidateChatRequestFilter : RequestFilter {
    override fun supports(request: ChatRequest): Boolean = request.message.isBlank()

    override fun apply(request: ChatRequest): ChatRequest {
        throw IllegalArgumentException("Chat message must not be blank")
    }
}
